### PR TITLE
Converted ModelPlugin to interface for testability; added mock entiti…

### PR DESCRIPTION
**Pull request by tomikazi**
…es to get tests

### CODE REVIEW

File: pkg/northbound/gnmi/v2/get.go [hunk 1]
```diff
@@ -154,8 +154,8 @@ func (s *Server) checkOpaAllowed(ctx context.Context, targetInfo targetInfo, con
 		return nil, err
 	}
 
-	targetVersion := filterTargetForURL(modelPlugin.Info.Version)
-	targetType := filterTargetForURL(modelPlugin.Info.Name)
+	targetVersion := filterTargetForURL(modelPlugin.GetInfo().Info.Version)
```
Comment (adibrastegarnia) [inline pkg/northbound/gnmi/v2/get.go:157]: I think this function somehow redundant isn't it? GetInfo then info?

File: pkg/northbound/gnmi/v2/get_test.go [hunk 2]
```diff
@@ -31,16 +31,18 @@ import (
...
 	return &Server{
 		mu:             sync.RWMutex{},
-		pluginRegistry: gnmitest.NewMockPluginRegistry(mctl),
-		topo:           gnmitest.NewMockStore(mctl),
+		pluginRegistry: registryMock,
+		topo:           topoMock,
 		transactions:   txStore,
 		configurations: cfgStore,
-	}, mctl, test
+	}, mctl, test, topoMock, registryMock
```
Comment (adibrastegarnia) [inline pkg/northbound/gnmi/v2/get_test.go:45]: I would suggest to define an struct and include all these mocks and information that should be returned to the caller.

**Review by teone — APPROVED** [review]

File: pkg/northbound/gnmi/v2/get.go (hunk 1 above)
Comment (tomikazi) [reply pkg/northbound/gnmi/v2/get.go:157]: No... GetInfo returns ModelPluginInfo, which is a record (previously named ModelPlugin) which contains another structure defined by API called ModelInfo. Yes, it looks somewhat odd, but I think we should leave it as is.

File: pkg/northbound/gnmi/v2/get_test.go (hunk 2 above)
Comment (tomikazi) [reply pkg/northbound/gnmi/v2/get_test.go:45]: Yeah, I was already working towards that in the current PR.

**Pull request merged**